relayer/chains/mock/message_handlers.go

```diff
@@ -0,0 +1,90 @@
+package mock
+
+import (
+	"fmt"
+
+	chantypes "github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
+	"github.com/cosmos/relayer/v2/relayer/ibc"
+	"github.com/cosmos/relayer/v2/relayer/provider"
+	"github.com/cosmos/relayer/v2/relayer/provider/cosmos"
+	"go.uber.org/zap"
+)
+
+type MsgHandlerParams struct {
+	mcp           *MockChainProcessor
+	PacketInfo    *chantypes.Packet
+	FoundMessages map[ibc.ChannelKey]map[string]map[uint64]provider.RelayerMessage
+}
+
+var messageHandlers = map[string]func(MsgHandlerParams){
+	ibc.MsgTransfer:        handleMsgTransfer,
+	ibc.MsgRecvPacket:      handleMsgRecvPacket,
+	ibc.MsgAcknowledgement: handleMsgAcknowlegement,
+
+	// TODO handlers for packet timeout, client, channel, and connection messages
+}
+
+func retainMessage(p MsgHandlerParams, channelKey ibc.ChannelKey, message string, sequence uint64, ibcMessage provider.RelayerMessage) {
+	if _, ok := p.FoundMessages[channelKey]; !ok {
+		p.FoundMessages[channelKey] = make(map[string]map[uint64]provider.RelayerMessage)
+	}
+	if _, ok := p.FoundMessages[channelKey][message]; !ok {
+		p.FoundMessages[channelKey][message] = make(map[uint64]provider.RelayerMessage)
+	}
+	p.FoundMessages[channelKey][message][sequence] = ibcMessage
```

Comment:
No chance we are ever calling `retainMessage()` from concurrent goroutines? 
It didn't look like that is the case here but something to keep in mind as we flesh out this design because this may require a lock or other synchronization method.

Reply:
There isn't parallelization yet but I have a comment in the mock chain processor about how this could be parallelized.